src/codegen/templates/property_getter_template.py.mako

```diff
@@ -0,0 +1,165 @@
+<%def name="script_property_getter(attribute)">\
+<%
+        from codegen.utilities.text_wrappers import wrap, docstring_wrap
+    %>\
+    @property
+    def ${attribute.name}(self):
+        """
+        ${attribute.description | docstring_wrap(8, 12)}
+        """
+## Script instantiation of numpy arrays for input parameters that are lists, and ctypes variables for
+## output parameters that will be passed by reference.
+<%
+        instantiation_line = None
+        if not attribute.has_explicit_read_buffer_size:
+            instantiation_line = 'val = {0}()'.format(attribute.ctypes_data_type)
+    %>\
+    %if instantiation_line:
+        ${instantiation_line}
+
+    %endif
+\
+    %if len(attribute.c_function_name) < 33:
+        cfunc = lib_importer.${'windll' if attribute.calling_convention=='StdCall' else 'cdll'}.DAQmxGet${attribute.c_function_name}
+    %else:
+        cfunc = (lib_importer.${'windll' if attribute.calling_convention=='StdCall' else 'cdll'}.
+                 DAQmxGet${attribute.c_function_name})
+    %endif
+\
+## Create argument ctypes types list.
+    %if attribute.calling_convention == 'StdCall':
+<%
+            argtypes = []
+            for handle_parameter in attribute.handle_parameters:
+                if handle_parameter.ctypes_data_type == 'ctypes.c_char_p':
+                    argtypes.append('ctypes_byte_str')
+                else:
+                    argtypes.append(handle_parameter.ctypes_data_type)
+
+            if (attribute.is_list and attribute.ctypes_data_type != 'ctypes.c_char_p' and
+                    attribute.bitfield_enum is None):
+                argtypes.append("wrapped_ndpointer(dtype={0}, flags=('C','W'))"
+                                .format(attribute.ctypes_data_type))
+            elif attribute.ctypes_data_type == 'ctypes.c_char_p':
+                argtypes.append(attribute.ctypes_data_type)
+            else:
+                argtypes.append('ctypes.POINTER({0})'.format(attribute.ctypes_data_type))
+
+            if attribute.has_explicit_read_buffer_size:
+                argtypes.append('ctypes.c_uint')
```

Comment:
Same comments here from above. Moving this logic to the `attribute` class could clean this up a bit!